### PR TITLE
pulseaudio: update to 15.0.

### PR DESCRIPTION
**Pull request by steinex**
* Enabled elogind functionality, ~so needs testing without elogind.~
* ~Build on arm fails due to some long unsigned int vs. size_t issues which I would need help with.~
* core-util-test still fails on CI

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
The actual error is

```
../src/pulsecore/remap_neon.c: In function 'remap_ch4_s16ne_neon':
../src/pulsecore/remap_neon.c:234:9: error: inconsistent operand constraints in an 'asm'
  234 |         __asm__ __volatile__ (
      |         ^
../src/pulsecore/remap_neon.c:256: confused by earlier errors, bailing out
lto-wrapper: fatal error: /usr/bin/armv7l-linux-gnueabihf-gcc returned 1 exit status
```

Maybe try disabling lto?

**Comment by ericonr** [comment]
What does the elogind functionality bring? Just to have an idea if enabling it is really worth it.

**Comment by steinex** [comment]
> Maybe try disabling lto? Unlikely to actually fix it, just a first step in debugging.

That actually fixed it. I disabled LTO for `arm*` now.

> What does the elogind functionality bring? Just to have an idea if enabling it is really worth it.

It gives each login session a PA client. I'm not sure what that actually means or what it's good for, however you added this comment in the last update: "`XXX: new version should be able to enable systemd functionality using elogind`".

I would be fine to leave it disabled.

**Comment by steinex** [comment]
I just got told from upstream what this does is it prevents pulseaudio from auto-exiting during a login session. Normally PA exits if there are no clients in a while.

File: srcpkgs/pulseaudio/template [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
 build_style=meson
-# XXX: new version should be able to enable systemd functionality using elogind
 configure_args="-Djack=enabled -Dlirc=disabled -Dhal-compat=false -Dorc=enabled
  -Dgtk=disabled -Dsystemd=disabled -Dwebrtc-aec=enabled
- -Dbluez5=true -Dbluez5-ofono-headset=false -Dbluez5-native-headset=true
- -Dudevrulesdir=/usr/lib/udev/rules.d
+ -Dbluez5=enabled -Dbluez5-ofono-headset=false -Dbluez5-native-headset=true
+ -Delogind=enabled -Dudevrulesdir=/usr/lib/udev/rules.d
  -Dbashcompletiondir=/usr/share/bash-completion/completions"
 hostmakedepends="cmake m4 gettext libtool orc-devel perl-XML-Parser pkg-config"
```
Comment (abenson) [inline srcpkgs/pulseaudio/template:11]: Test build failed for me, needs `doxygen`?

**Pull request closed without merging**